ci/ci/github.py

```diff
@@ -442,11 +468,20 @@ def github_status(self):
             return
 
         if (not self.batch or
-                (on_deck and self.batch.attributes['target_sha'] != self.target_branch.sha)):
-
+                (on_deck and self.batch.attributes['target_sha'] != self.target_branch.sha) or
+                self.retry_requested):
             if on_deck or self.target_branch.n_running_batches < 8:
                 self.target_branch.n_running_batches += 1
                 async with repos_lock:
+                    if self.retry_requested:
+                        log.info(f'retrying {self.number} with existing batch {self.batch}')
+                        if self.batch is not None:
+                            log.info(f'cancel batch {self.batch.id} because a retry was requested')
+                            await self.batch.cancel()
+                            self.batch = None
+                            self.build_state = None
+                            self.retry_requested = False
```

Comment:
So I think this can be simplified: just increment the attempt_id no retry.  Then the attempt filter will return no batch, a new one will be started, and the old batches will get cancelled by the existing cancellation logic.